R/model-wrappers.R

```diff
@@ -290,3 +290,173 @@ forecastHybrid_model <- function(y = NULL, samples = NULL,
   return(sample_from_model)
 
 }
+
+
+
+#' Stack models according to CRPS
+#'
+#' @description
+#' Provides a wrapper for different EpiSoon models and generates
+#' a mixture model of these models based on the (Continuous) Rank Probability
+#' Score
+#'
+#' A list of models is supplied. These models are fit to the data up until a
+#' period of observations of size `weighting_period`. Forecasts are generated
+#' from all the models for all time points in the `weighting_period`.
+#' Predictive samples generated by the individual models are then used to
+#' create model weights in an ensemble based on CRPS. All models are then
+#' refitted for the entire timeseries and predictions are generated from these
+#' models. Draws from the individual model predictive samples are then used
+#' to generate a mixture model with the weights obtained in the previous step.
+#'
+#' The weights are computed using \code{\link[stackr]{stack_crps}} from
+#' the package `stackr` to minimise CRPS. The function
+#' \code{\link[stackr]{mixture_from_sample}} from the same package is used
+#' to draw samples from the
+#' individual models to form the mixture models.
+#'
+#' @param models A list of models. Models must be analogous to the form
+#' `function(...){EpiSoon::fable_model(model = , ...)}` or
+#' `function(...){EpiSoon::bsts_model(model = , ...)}`.
+#' @inheritParams bsts_model
+#' @param weighting_period The number of most recent timepoints to hold out to
+#' generate the weights for the mixture model
+#' @return A dataframe of predictions (with columns representing the
+#'  time horizon and rows representing samples).
+#' @export
+#' @importFrom purrr map_dfr
```

Comment:
this is a suggested fn so perhaps should not be importing anything

Reply:
so just delete all the imports?

Reply:
yes

Reply:
and add the packages it needs as suggests if not already improted